clients/vscode/src/chat/ChatViewProvider.ts

```diff
@@ -19,27 +20,26 @@ export class ChatViewProvider implements WebviewViewProvider {
 
   public async resolveWebviewView(webviewView: WebviewView) {
     this.webview = webviewView;
+    this.isReady = this.agent.status === "ready";
     const extensionUri = this.context.extensionUri;
 
     webviewView.webview.options = {
       enableScripts: true,
       localResourceRoots: [extensionUri],
     };
-    // FIXME: we need to wait for the server to be ready, consider rendering a loading indicator
+
+    // FIXME: does 1s delay needed?
     if (this.agent.status !== "ready") {
       await new Promise<void>((resolve) => {
-        this.agent.on("didChangeStatus", (status) => {
-          if (status === "ready") {
-            resolve();
-          }
-        });
+        setTimeout(() => {
+          this.isReady = this.agent.status === "ready";
+          resolve();
+        }, 1000);
```

Comment:
@icycodes I don't know how to handle this case
If I don't add the 1s delay, this.isReady will always be `false` and I can't get any event in `this.agent.on("didChangeStatus"`

Reply:
I'm not sure, but it may be related to didChangeStatus listener is added after an await fetchSeverInfo

Reply:
Removed the 1s delay, add `fetchServerInfo` when ready
```
private async renderChatPage(serverInfo?: ServerInfo) {
    if (!serverInfo) {
      serverInfo = await this.agent.fetchServerInfo();
    }
    if (this.webview) {
      this.webview.webview.html = this.getWebviewContent(serverInfo);
    }
  }
```